app/models/piece.rb

```diff
@@ -5,7 +5,52 @@ class Piece < ActiveRecord::Base
   validates :color, inclusion: { in: %w(black white) }
   validates :type, inclusion: { in: %w(Pawn Rook Bishop Knight King Queen) }
 
-end
+  private
+
+  #ALL MOVE CHECKS ASSUME WHITE PLAYER IS AT BOTTOM OF 2-D ARRAY/BOARD, AND BLACK PLAYER IS ON TOP.  NO IMPACT ON VIEW. -Chad
+
+  def x_distance(x) #calculates how many tiles piece has moved on x-axis
+    (x_position - x).abs
+  end
+
+  def y_distance(y) #calculates how many tiles piece has moved on y-axis
+    (y_position - y).abs
+  end
+
+  def moved?(x,y) #validates whether piece is at its origin position
+    x != x_position || y != y_position
+  end
+
+  def horizontal_or_vertical_move?(x,y) #validates a move any amount of spaces along one axis
+    (x_distance(x) == 0 || y_distance(y) == 0)
+  end
 
-# Sample piece creation - Piece.create(type:"Queen", color:"white") type must be capitalized.
-                      # - Rook.create(color:"white")
\ No newline at end of file
+  def diagonal_move?(x,y) #validates a move any amount of spaces along a diagonal
```

Comment:
Could you possibly return either your x or y distance (known to be equal here), then just do the final eval in king? This would allow you to get rid of `radial_move`